spec/lib/pre_assembly/batch_spec.rb

```diff
@@ -218,33 +262,60 @@
   end
 
   describe '#load_checksums' do
-    it 'loads checksums and attach them to the ObjectFiles' do
-      multimedia.send(:all_object_files).each { |f| expect(f.checksum).to be_nil }
-      multimedia.digital_objects.each { |dobj| multimedia.send(:load_checksums, dobj) }
-      multimedia.send(:all_object_files).each { |f| expect(f.checksum).to match(md5_regex) }
+    it 'loads checksums and attaches them to the ObjectFiles' do
+      multimedia.digital_objects.each do |dobj|
+        dobj.object_files.each { |f| expect(f.checksum).to be_nil }
+        batch.send(:load_checksums, dobj)
+        dobj.object_files.each { |f| expect(f.checksum).to match(md5_regex) }
+      end
     end
   end
 
   describe '#un_pre_assembled_objects' do
     it 'returns all objects if there are no pre_assembled_object_containers' do
-      expect(flat_dir_images.un_pre_assembled_objects).to eq(flat_dir_images.digital_objects)
+      all_dobj_array = flat_dir_images.digital_objects.to_a
+      un_preassembled_dobj_array = flat_dir_images.un_pre_assembled_objects.to_a
+      un_preassembled_dobj_array.each_with_index do |dobj, i|
+        # Each time it's called, #to_a iterates over the Enumerable and re-instantiates the yeilded
+        # objects anew.  DigitalObject doesn't have an == method, so compare the relevant fields here
+        # since default == will just check whether the two are the same instance.
+        %i[batch container stageable_items object_files label pid source_id stager].all? do |field_name|
+          expect(dobj.send(field_name)).to eq(all_dobj_array[i].send(field_name))
+        end
```

Comment:
my first instinct was to add an `#==` method to `DigitalObject` so that this array comparison would work implicitly and less verbosely.  the comparison used here is the comparison i initially used for that.  but i wasn't sure it was the right comparison and it was only being used implicitly by a couple tests in this file.  so i went with the more verbose thing here for now, and split the `DigitalObject#==` implementation into its own branch, which i'll put up as a separate PR for consideration.